ir/src/transition_constraints/graph.rs

```diff
@@ -232,6 +300,9 @@ impl Node {
 #[derive(Debug, Eq, PartialEq)]
 pub enum Operation {
     Const(u64),
+    ScalarConstant(u64),
+    VectorConstant(VectorAccess),
+    MatrixConstant(MatrixAccess),
```

Comment:
Thanks. I made a slight variation and went with using `VectorAccess` and `MatrixAccess` structs as arguments.
```
pub enum ConstantValue {
    Inline(u64),
    Scalar(String),
    Vector(VectorAccess),
    Matrix(MatrixAccess),
}
```

@grjte had a slight preference of using structs here instead of fixing index order. Please let me know if you prefer to use struct fields as arguments instead or if there are any performance implications in using struct here (I couldn't find anything on that).

Reply:
I think this works. In general, we are much less concerned about performance here - so, w/e makes the code more clear is usually better.

It does feel a bit weird that we use structs from the parser in the IR, but I guess we do it in a few other places already. Maybe there should be a `air-script-core` or `air-script-common` crate which defines such commonly used structs.

Reply:
Makes sense. I've opened #98 for this.